test/e2e/util/hypershift_framework.go

```diff
@@ -277,56 +277,57 @@ func (h *hypershiftTest) createHostedCluster(opts *core.CreateOptions, platform
 	return hc
 }
 
-// NOTE: teardownHostedCluster shouldn't include any subtests
-// as it is called during the cleanup phase where t.Run() is not supported.
-func (h *hypershiftTest) teardownHostedCluster(ctx context.Context, hc *hyperv1.HostedCluster, opts *core.CreateOptions, artifactDir string) {
+// NOTE: teardownHostedCluster shouldn't start any subtests with t.Run() when cleanupPhase=True, this is not a supported operation and will fail immediately
+func teardownHostedCluster(t *testing.T, ctx context.Context, hc *hyperv1.HostedCluster, client crclient.Client, opts *core.CreateOptions, artifactDir string, cleanupPhase bool) {
 	// TODO (Mulham): dumpCluster() uses testName to construc dumpDir, since we removed sub tests from this function
```

Comment:
Is this still relevant and could we do it as part of this pr or is it bigger piece of work?

Reply:
hmm, not sure if we still need this.
Basically we get 1 dump currently before we destroy the cluster and save it to `<artifactDir>/<testName>`. but if the rare cases, where the destroy fails, another dump is generated, but currently without this, its saved in the same path overriding the first one.

Reply:
I am not sure whether we need all those different dumps to debug any issue.

Reply:
I think it's nice to have if we ever need it but not a priority. We can create a Jira for this and stick it in the backlog, wdyt?

Reply:
sounds good